src/test/java/com/github/jreddit/action/SubmitActionsTest.java

```diff
@@ -1,5 +1,313 @@
 package com.github.jreddit.action;
 
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.json.simple.JSONObject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import static org.junit.Assert.assertEquals;
+
+import com.github.jreddit.entity.User;
+import com.github.jreddit.utils.ApiEndpointUtils;
+import com.github.jreddit.utils.restclient.Response;
+import com.github.jreddit.utils.restclient.RestClient;
+import com.github.jreddit.utils.restclient.RestResponse;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.apache.http.HttpResponse;
+
 public class SubmitActionsTest {
 
+    public static final String COOKIE = "cookie";
+    public static final String REDDIT_NAME = "all";
+    public static final String USERNAME = "TestUser";
+    public static final String MODHASH = "modhash";
+    private SubmitActions submitAction;
+    private RestClient restClient;
+    private User user;
+    private Response responseNotAuthor;
+    private Response responseUserRequired;
+    private Response responseEmpty;
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    /**
+     * Mock classes it depends on.
+     */
+    @Before
+    public void setup() {
+        user = mock(User.class);
+        when(user.getCookie()).thenReturn(COOKIE);
+        when(user.getModhash()).thenReturn(MODHASH);
+        restClient = mock(RestClient.class);
+        submitAction = new SubmitActions(restClient, user);
+
+        JSONObject errorObject = new JSONObject();
+        errorObject.put("call", ".error.NOT_AUTHOR.field-thing_id");
+        responseNotAuthor = new RestResponse("test", errorObject, mock(HttpResponse.class));
+
+        errorObject = new JSONObject();
+        errorObject.put("call", ".error.USER_REQUIRED");
+        responseUserRequired = new RestResponse("test", errorObject, mock(HttpResponse.class));
+
+        errorObject = new JSONObject();
+        responseEmpty = new RestResponse("", errorObject, mock(HttpResponse.class));
+    }
+
+    /**
+     * Test deleting a thing.
+     */
+    @Test
+    public void testDelete() {
+        // Stub REST client methods
+        when(user.getModhash()).thenReturn(MODHASH);
+        when(user.getCookie()).thenReturn(COOKIE);
+        when(restClient.post(
+                "id=" + "fullname" + "&uh=" + user.getModhash(),
+                ApiEndpointUtils.DELETE, user.getCookie()))
+                .thenReturn(responseEmpty);
+
+        // Delete thing succesfully
+        assertEquals(submitAction.delete("fullname"), true);
```

Comment:
You can use assertTrue() or asserFalse() for simplicity. Can you do for the rest, as well? Don't want to leave you a comment for each of them :)